tf_quant_finance/volatility/heston_approximation.py

```diff
@@ -0,0 +1,239 @@
+# Lint as: python3
+# Copyright 2020 Google LLC
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import tensorflow as tf
+
+import math
+import tf_quant_finance as tff
+
+"""Method for semi-analytical Heston option price.
+
+Heston originally published in 1993 his eponymous model. He provided a semi-
+analytical formula for pricing European option via Fourrier transform under his
+model. However, as noted by Albrecher, the characteric function used in Heston 
+paper can suffer numerical issues because of the discontinuous nature of the 
+square root function in the complex plane, and a second version of the 
+characteric function which doesn't suffer this shortcoming should be used 
+instead. Attari further refined the numerical method by reducing the number of 
+numerical integrations (only one Fourrier transform instead of two) and with
+an integrand function with better convergence (converging in 1 / u **2 
+instead of 1 / u).  Attari's numerical method is implemented here.
+ 
+## References
+
+[1] Hansjorg Albrecher, The Little Heston Trap
+https://perswww.kuleuven.be/~u0009713/HestonTrap.pdf
+
+[2] Mukarram Attari, Option Pricing Using Fourier Transforms: A Numerically 
+Efficient Simplification
+https://papers.ssrn.com/sol3/papers.cfm?abstract_id=520042
+
+[3] Steven L. Heston, A Closed-Form Solution for Options with Stochastic 
+Volatility with Applications to Bond and Currency Options
+http://faculty.baruch.cuny.edu/lwu/890/Heston93.pdf
+"""
+
+def heston_price(*,
```

Comment:
Could you please create a folder in `models` and call it `heston`, like we did for hull white. Then move the heston model into that folder. Inside `heston` you should create `approximations` folder like we did for Black-Scholes (see [here](https://github.com/google/tf-quant-finance/tree/master/tf_quant_finance/black_scholes)). You should put this file there. Thanks!

Also, should we call this function `eu_option_price`? User then can access it via `tff.models.heston.approximations.eu_option_price`

Reply:
ok, I have renamed module/folder accordingly